rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/NodesRecoveryManager.java

```diff
@@ -279,4 +267,81 @@ private boolean isEligible(RMNode node) {
         return node != null && node.isFree() && !node.isLocked();
     }
 
+    private class RecoveryNode implements Node {
+
+        private String name;
+
+        private String url;
+
+        private RecoveryNode(String name, String url) {
+            this.name = name;
+            this.url = url;
+        }
+
+        @Override
+        public NodeInformation getNodeInformation() {
+            return new NodeInformation() {
+                @Override
+                public String getName() {
+                    return RecoveryNode.this.name;
+                }
+
+                @Override
+                public String getURL() {
+                    return RecoveryNode.this.url;
+                }
+
+                @Override
+                public VMInformation getVMInformation() {
+                    return null;
+                }
+            };
+        }
+
+        @Override
+        public VMInformation getVMInformation() {
+            return null;
+        }
+
+        @Override
+        public ProActiveRuntime getProActiveRuntime() {
+            return null;
+        }
+
+        @Override
+        public Object[] getActiveObjects() throws NodeException, ActiveObjectCreationException {
+            return new Object[0];
+        }
+
+        @Override
+        public Object[] getActiveObjects(String s) throws NodeException, ActiveObjectCreationException {
+            return new Object[0];
+        }
+
+        @Override
+        public int getNumberOfActiveObjects() throws NodeException {
+            return 0;
+        }
+
+        @Override
+        public void killAllActiveObjects() throws NodeException, IOException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)